src/_data/figma.js

```diff
@@ -0,0 +1,176 @@
+const CacheAsset = require('@11ty/eleventy-cache-assets');
+const fs = require('fs');
+const path = require('path');
+
+const BASE_URL = 'https://api.figma.com';
+const FIGMA_ACCESS_TOKEN = process.env.FIGMA_ACCESS_TOKEN;
+const FIGMA_FILE_KEY = process.env.FIGMA_FILE_KEY;
+
+/**
+ * Get list of components that are both in docs and in Figma
+ * @param {array} componentsInDocs
+ * @param {array} componentsInFigma
+ * @returns {array} Array of objects of Figma components
+ */
+const getMatchingComponents = (componentsInDocs, componentsInFigma) => {
+  if (!componentsInDocs || !componentsInFigma) return;
+
+  let matchingComponents = [];
+  for (let i = 0; i < componentsInDocs.length; i++) {
+    for (let j = 0; j < componentsInFigma.length; j++) {
+      if (componentsInDocs[i] === componentsInFigma[j].name.toLowerCase()) {
+        matchingComponents.push(componentsInFigma[j]);
+      }
+    }
+  }
+  return matchingComponents;
+};
+
+/**
+ * Get an list of components that has an Markdown file in the components folder
+ */
+const getComponentsFromDocs = () => {
+  return fs
+    .readdirSync('./src/components')
+    .filter((file) => path.extname(file).toLowerCase() === '.md' && !file.startsWith('_'))
+    .map((file) => file.split('.')[0]);
+};
+
+/**
+ * Get component nodes from Figma
+ */
+const getComponentsFromFigma = async () => {
+  try {
+    return await CacheAsset(`${BASE_URL}/v1/files/${FIGMA_FILE_KEY}`, {
+      duration: '1d',
+      type: 'json',
+      fetchOptions: {
+        method: 'GET',
+        headers: {
+          'X-Figma-Token': FIGMA_ACCESS_TOKEN,
+        },
+      },
+    }).then((res) => res.document.children); // document.children has all pages
+  } catch (e) {}
+};
+
+/**
+ * Create array of object with component name and Figma ID
+ * @param {array} components
+ * @returns {array}
+ */
+const getComponentFigmaIds = (components) => {
+  if (!components) return;
+
+  return components
+    .filter((component) => component.children.length)
+    .map((component) => {
+      return {
+        name: component.name.toLowerCase(),
+        id: component.children
+          .filter((component) => component.name.toLowerCase() === 'docs')
+          .reduce((acc, component) => {
+            if (component.id) return component.id;
+            return acc;
+          }, ''),
+      };
+    });
+};
+
+/**
+ * Fetch component URLs from Figma API
+ * @param {array} components
+ */
+const getComponentFigmaUrls = async (components) => {
+  if (!components) return;
+
+  const ids = Array.isArray(components)
+    ? components
+        .filter((component) => component.id)
+        .map((component) => component.id)
```

Comment:
I don't know if you like this style or not, but you can use destructuring to nice effect here:
```suggestion
        .filter(({id}) => id)
        .map(({id}) => id)
```